torch/csrc/autograd/variable.cpp

```diff
@@ -109,6 +111,22 @@ void Variable::Impl::detach_() {
   output_nr_ = 0;
 }
 
+void Variable::Impl::backward(
+    at::optional<Tensor> gradient,
+    bool keep_graph,
+    bool create_graph) {
+  std::vector<Edge> edges;
+  edges.emplace_back(grad_fn_, output_nr_);
+
+  std::vector<Variable> inputs;
+  if (!gradient.has_value()) {
+    gradient = make_variable(at::ones_like(data_), /*requires_grad=*/false);
+  }
+  inputs.push_back(std::move(as_variable_ref(*gradient)));
+
+  Engine::get_default_engine().execute(edges, inputs, keep_graph, create_graph);
```

Comment:
nit: would be simpler to just do `{gradient}` instead of constructing `inputs` above